### PR TITLE
Create Harvard-Stellenbosch-University.csl

### PR DESCRIPTION
**Pull request by Yusufras**

### CODE REVIEW

**Comment by Yusufras** [comment]
Need some help please.

File: Harvard-Stellenbosch-University.csl [hunk 1]
```diff
@@ -0,0 +1,309 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only" default-locale="en-GB">
+  <info>
+    <title>Harvard - Stellenbosch University  - Yusuf Ras</title>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: don't include your name in the title. This should just be: 
```Harvard - Stellenbosch University```

File: Harvard-Stellenbosch-University.csl [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    <id>http://csl.mendeley.com/styles/21406231/harvard-stellenbosch-university--2</id>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: You can see all the formal requirements [here](https://github.com/citation-style-language/styles/wiki/Style-Requirements). As per 5. this should be ```http://www.zotero.org/styles/harvard-stellenbosch-university```

File: Harvard-Stellenbosch-University.csl [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    <link href="http://www.zotero.org/styles/harvard-university-of-cape-town" rel="self"/>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: This should be:
```<link href="http://www.zotero.org/styles/harvard-stellenbosch-university" rel="self"/>```

File: Harvard-Stellenbosch-University.csl [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    <link href="http://www.zotero.org/styles/harvard-university-of-gloucestershire" rel="template"/>
+    <link href="http://www.lib.uct.ac.za/wp-content/uploads/2014/02/harvard-uct-2014.pdf" rel="documentation"/>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: we'd want the actual link to the styleguide here. This is the one for capetown.

File: Harvard-Stellenbosch-University.csl [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+    <author>
+      <name>Yusuf Ras</name>
+      <email>yr@sun.ac.za</email>
+    </author>
+    <contributor>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: no need to list yourself twice: author is enough.

File: Harvard-Stellenbosch-University.csl [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+    <category citation-format="author-date"/>
+    <category field="generic-base"/>
+    <updated>2015-09-01T08:01:34+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <locale xmlns="http://purl.org/net/xbiblio/csl" xml:lang="en-GB">
+    <terms>
+      <term name="volume" form="short">v.</term>
+      <term name="available at">available</term>
+      <term name="open-quote">�</term>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: something went wrong here. What type of quotation marks do you need?

File: Harvard-Stellenbosch-University.csl [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+            <text variable="title" quotes="true" text-case="capitalize-first"/>
+          </else>
+        </choose>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-short">
+    <names variable="author">
+      <name form="short" and="symbol" delimiter-precedes-last="never" et-al-min="5" et-al-use-first="5" et-al-subsequent-min="3" et-al-subsequent-use-first="1" initialize-with="." sort-separator=" "/>
+      <et-al suffix="., " font-style="italic"/>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: while possible, don't set a suffix on et al.: the period is there anyway, and the comma should probably be after the name either way, no? So you could just set it after the macro in the citation.

Also, ```el-al-use-first``` is always at least one less than ```et-al-min```. And could you confirm you really want (Smith, Meyer, Jones & Davis et al. 2001)? Four authors before et al seems highly unusual.

File: Harvard-Stellenbosch-University.csl [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="access">
+    <choose>
+      <if variable="DOI"/>
```
Comment (adam3smith) [inline Harvard-Stellenbosch-University.csl]: this seems odd: print nothing at all if you have a DOI? Are you sure about this?

**Comment by adam3smith** [comment]
Also, the filename for your style should be
```harvard-stellenbosch-university.csl```
You can update your pull request by clicking the pencil icon here:
https://github.com/Yusufras/styles/blob/patch-6/Harvard-Stellenbosch-University.csl
(please do not create any additional pull requests. We can fix everything in this one).
Let us know when you're done, we don't get notifications.

**Comment by rmzelle** [comment]
There are a few things left to fix.

First, you need to change the file name of the style to "harvard-stellenbosch-university.csl". See https://help.github.com/articles/renaming-a-file/ for instructions.

Second, you need to replace

```xml
    <author>
      <name>Yusuf Ras</name>
      <email>yr@sun.ac.za</email>
    </author>
    <contributor>      
    <uri>http://www.mendeley.com/profiles/yusuf-ras/</uri>
    </contributor>
```

by

```xml
    <author>
      <name>Yusuf Ras</name>
      <email>yr@sun.ac.za</email>
      <uri>http://www.mendeley.com/profiles/yusuf-ras/</uri>
    </author>
```

Third, the current version has two "self" links. Remove the line with `<link href="http://www.zotero.org/styles/harvard-university-of-cape-town" rel="self"/>`.

**Comment by rmzelle** [comment]
You currently have

```
    <author>
    <name>Yusuf Ras</name>
    <email>yr@sun.ac.za</email>
    <uri>http://www.mendeley.com/profiles/yusuf-ras/</uri>
    </author>
    <uri>http://www.mendeley.com/profiles/yusuf-ras/</uri>
    </contributor>
```

You still need to delete the last two lines.

**Comment by Yusufras** [comment]
Comments please

**Comment by Yusufras** [comment]
When will this be available on the style repository?

**Comment by rmzelle** [comment]
One last thing that strikes me as odd is how you redefined the quotation marks:

```
      <term name="open-quote">""</term>
      <term name="close-quote">""</term>
      <term name="open-inner-quote">""</term>
      <term name="close-inner-quote">""</term>
```

If you look at https://github.com/citation-style-language/locales/blob/master/locales-en-GB.xml#L64, you can see that normally, we only have a single quotation mark for each term. With your code, you should be getting output like `""text""` with duplicated quote marks. Are you sure you wouldn't want to be using

```
    <term name="open-quote">“</term>
    <term name="close-quote">”</term>
    <term name="open-inner-quote">‘</term>
    <term name="close-inner-quote">’</term>
```

?

**Comment by adam3smith** [comment]
I also still don't see a valid documentation link. You're currently linking to a UCT styleguide which isn't even there any more.

**Comment by Yusufras** [comment]
Can i use the one you recommend. Thank you
    <term name="open-quote">“</term>
    <term name="close-quote">”</term>
    <term name="open-inner-quote">‘</term>
    <term name="close-inner-quote">’</term>

**Comment by Yusufras** [comment]
I need help with this. I have edited the UCT style. What is the documentation link and where do i get it?

**Comment by rmzelle** [comment]
It doesn't seem like you change the quote terms yet. You can do this via the pencil icon at https://github.com/Yusufras/styles/blob/patch-6/harvard-stellenbosch-university.csl.

As for the "documentation" link, your style currently includes `<link href="http://www.lib.uct.ac.za/wp-content/uploads/2014/02/harvard-uct-2014.pdf" rel="documentation"/>`. This link points to the website of the University of Cape Town, and results in a 404 error. Assuming that the Stellenbosch citation format guidelines are available online, this link should be updated to point to those.

**Comment by Yusufras** [comment]
I have done the changes as recommended. Please advice.

**Comment by rmzelle** [comment]
Should the link be to http://libguides.sun.ac.za/content.php?pid=344501&sid=2818229? I get an error with your link.

Also note that you need to escape ampersands in XML (using "&amp;" instead of "&"). So e.g. http://libguides.sun.ac.za/content.php?pid=344501&sid=2818229 should be formatted as `<link href="http://libguides.sun.ac.za/content.php?pid=344501&amp;sid=2818229" rel="documentation"/>`.

**Comment by Yusufras** [comment]
I have done the corrections. Yes yours is the correct link. I put the WWW in front which is probably incorrect. Is this working fine know?

**Comment by rmzelle** [comment]
Great, thanks for contributing! Sorry for let you wait for a few days!

**Comment by Yusufras** [comment]
No thank you, without your help it would not have been possible.

Vriendelike groete / Kind regards
Yusuf Ras

Fakulteitsbibliotekaris AgriWetenskappe / Faculty Librarian AgriSciences
Biblioteek- & Inligtingsdiens / Library & Information Service
Universiteit Stellenbosch University  | map<https://www.google.co.za/maps/place/JS+Gericke+Library/@-33.932975,18.8640494,377m/data=!3m1!1e3!4m2!3m1!1s0x1dcdb25d82940c29:0x80f6dcd585958374> | Tel: +27 (021) 808 4345 | yr@sun.ac.za<mailto:yr@sun.ac.za>

**Comment by Yusufras** [comment]
How do I add this style to Mendeley so it is available to all on campus?

Vriendelike groete / Kind regards
Yusuf Ras

Fakulteitsbibliotekaris AgriWetenskappe / Faculty Librarian AgriSciences
Biblioteek- & Inligtingsdiens / Library & Information Service
Universiteit Stellenbosch University  | map<https://www.google.co.za/maps/place/JS+Gericke+Library/@-33.932975,18.8640494,377m/data=!3m1!1e3!4m2!3m1!1s0x1dcdb25d82940c29:0x80f6dcd585958374> | Tel: +27 (021) 808 4345 | yr@sun.ac.za<mailto:yr@sun.ac.za>

**Comment by rmzelle** [comment]
Mendeley supposedly updates their styles once every 24 hours, so check again later today or tomorrow and it should be available from within the app.

**Comment by Yusufras** [comment]
What do i need to do to edit the style. I need to add this to the style when using electronic resources.

Petrić, B. 2007. Rhetorical functions of citations in high- and low-rated master's theses. Journal of English for Academic Purposes 6(3). [Online], Available: http://www.sciencedirect.com/science [2007, November 30].

**Comment by adam3smith** [comment]
another pull request. If you click on the pencil icon at the top of the style in (our) style repository, it'll automatically fork it to a new branch for you and you can edit it there and the issue the pull request like you did here. (or are you asking for help in editing the style specifics?)

**Comment by Yusufras** [comment]
I don not see the pencil as indicated. I need to add this:  [Online], Available: http://www.sciencedirect.com/science [2007, November 30]. 
How do i go about this and where do i add it?

**Comment by adam3smith** [comment]
You'd definitely see a pencil icon here: https://github.com/citation-style-language/styles/blob/master/harvard-stellenbosch-university.csl at the top right, next to history.

**Comment by Yusufras** [comment]
So how and where do i add the above to the style?

**Comment by rmzelle** [comment]
> I need to add this to the style

It's easier for us to give you advice when you clearly show what "*this*" is. In addition to the desired output you showed in https://github.com/citation-style-language/styles/pull/1671#issuecomment-144982839, can you also show us what the style currently produces?

**Comment by Yusufras** [comment]
This how it looks like now:
Cai, J. & Leung, P. 2008. Towards a more general measure of revealed comparative advantage variation. Applied Economics Letters. 15(9):723–726. 

This is how it needs to look like:
Petrić, B. 2007. Rhetorical functions of citations in high- and low-rated master's theses. Journal of English for Academic Purposes 6(3). [Online], Available: http://www.sciencedirect.com/science [2007, November 30].

It basically got to do with date access. This is the format for any electronic item, article, electronic book, webpage etc.

**Comment by Yusufras** [comment]
If i add the below to the style, will it show this: Petrić, B. 2007. Rhetorical functions of citations in high- and low-rated master's theses. Journal of English for Academic Purposes 6(3). [Online], Available: http://www.sciencedirect.com/science [2007, November 30].
<macro name="access">
    <choose>
      <if variable="URL">
        <text value="[Online], Available:" suffix=" "/>
        <text variable="URL"/>
        <group prefix=" [" suffix="]">
          <date variable="accessed">
            <date-part name="year" suffix=" , "/>
            <date-part name="month" suffix="  "/>
            <date-part name="day"/>
          </date>
        </group>
      </if>
    </choose>
  </macro>

**Comment by rmzelle** [comment]
Ah, so it looks like you were able to figure this out by yourself? You can modify the existing copy of the style in the repository by using the pencil (edit) icon at https://github.com/citation-style-language/styles/blob/master/harvard-stellenbosch-university.csl

**Pull request merged**